Modules/Server/Sources/PocketCastsServer/Public/Search/EpisodeSearchTask.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+public struct EpisodeSearchEnvelope: Decodable {
+    public let episodes: [EpisodeSearchResult]
+}
+
+public struct EpisodeSearchResult: Decodable {
+    public let uuid: String
+    public let title: String
+    public let publishedDate: Date
+    public let duration: Double?
+    public let podcastUuid: String
+    public let podcastTitle: String
+}
+
+public class EpisodeSearchTask {
+    var session = URLSession.shared
+
+    public init() {}
+
+    public func search(term: String) async throws -> [EpisodeSearchResult] {
+        let searchURL = URL(string: "\(ServerConstants.Urls.cache())episode/search")!
+        var request = URLRequest(url: searchURL)
+        request.httpMethod = "POST"
+
+        let json: [String: Any] = ["term": term]
+
+        let jsonData = try? JSONSerialization.data(withJSONObject: json)
```

Comment:
Since this call throws, should we remove the `?`?
```suggestion
        let jsonData = try JSONSerialization.data(withJSONObject: json)
```